src/components/dropdown-filter/__spec__.js

```diff
@@ -72,6 +72,26 @@ describe('DropdownFilter', () => {
     });
   });
 
+  describe('componentWillReceiveProps', () => {
+    describe('should update the filter if visible value changed,', () => {
+      it('sets openingList to true', () => {
+        const props = {
+            name:'foo',
+            options: Immutable.fromJS([]),
+            visibleValue: 'abc',
+            freetext: true
+        }
+        instance = TestUtils.renderIntoDocument(
+          <DropdownFilter {...props} />
+        );
+
+        expect(instance.state.filter).toEqual('abc');
+        instance.componentWillReceiveProps({ ...props, visibleValue: 'def' });
+        expect(instance.state.filter).toEqual('def');
+      });
+    });
+  });
```

Comment:
Component's internal implementation shouldn't be tested.
You could use Enzyme `shallow` or `mount` wrapper and the `setProps` method instead of directly triggering `componentWillReceiveProps` on an instance. You should also check something that depends on the state instead of the state directly on an instance.